### ISSUE TITLE
DevTools loading with a blank white page

### ISSUE THREAD

**Issue by luisgustavoo** [flutter/devtools#2759]
![Screenshot_1](https://user-images.githubusercontent.com/59834387/110015693-ec222780-7d02-11eb-81d1-8efa20617b30.png)

C:\src\flutter\bin\flutter.bat doctor --verbose
[√] Flutter (Channel stable, 2.0.0, on Microsoft Windows [versÃ£o 10.0.19042.804], locale pt-BR)
    • Flutter version 2.0.0 at C:\src\flutter
    • Framework revision 60bd88df91 (26 hours ago), 2021-03-03 09:13:17 -0800
    • Engine revision 40441def69
    • Dart version 2.12.0

[√] Android toolchain - develop for Android devices (Android SDK version 30.0.2)
    • Android SDK at C:/Users/desenv2/AppData/Local/Android/Sdk
    • Platform android-30, build-tools 30.0.2
    • ANDROID_HOME = C:/Users/desenv2/AppData/Local/Android/Sdk
    • Java binary at: C:\Program Files\Android\Android Studio\jre\bin\java
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b01)
    • All Android licenses accepted.

[√] Android Studio (version 4.1.0)
    • Android Studio at C:\Program Files\Android\Android Studio
    • Flutter plugin can be installed from:
       https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
       https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b01)

[√] VS Code (version 1.53.2)
    • VS Code at C:\Users\desenv2\AppData\Local\Programs\Microsoft VS Code
    • Flutter extension version 3.20.0

[√] Connected device (1 available)
    • Android SDK built for x86 64 (mobile) • emulator-5554 • android-x64 • Android 10 (API 29) (emulator)

• No issues found!
Process finished with exit code 0

**Comment by devoncarew** [comment]
cc @kenzieschmoll

**Comment by devoncarew** [comment]
Thanks for the report! Can you asset not found path a bit on the right? Or, copy the path in here.

**Comment by kenzieschmoll** [comment]
Interesting. I just tried to reproduce this and I can see the error in the console, but I still see the codicons loaded properly in the app.
![Screen Shot 2021-03-04 at 11 23 34 AM](https://user-images.githubusercontent.com/43759233/110018412-1b716e00-7cdc-11eb-949b-1bae5c30db15.png)

It looks like we are try to load the font from two places:
1.  path `http://127.0.0.1:9100/assets/packages/codicon/lib/font/codicon.ttf`: fails
2. path `http://127.0.0.1:9100/assets/fonts/codicon.ttf`: succeeds

@devoncarew we are rewriting the pubspec with the changes in https://github.com/flutter/devtools/pull/2739 - any idea how we cold still be attempting to load the font from `package/codicon/lib`?

**Comment by luisgustavoo** [comment]
> Obrigado pelo relatório! Você consegue um caminho não encontrado do ativo um pouco à direita? Ou copie o caminho aqui.

`http://127.0.0.1:9100/assets/packages/codicon/lib/font/codicon.ttf`

**Comment by devoncarew** [comment]
I'm not sure why, but I do see the font listed in both places in the manaifests. I downloaded the tar.gz from pub and unpacked it.

devtools-2.0.0/build/assets/AssetManifest.json lhas entries for both `packages/codicon/lib/font/codicon.ttf` and `fonts/codicon.ttf`.

The same is try for devtools-2.0.0/build/assets/FontManifest.json. We don't package `assets/packages` (and I assume we don't want to). Here's the content:

```json
[
    {
        "family": "MaterialIcons",
        "fonts": [
            {
                "asset": "fonts/MaterialIcons-Regular.otf"
            }
        ]
    },
    {
        "family": "Roboto",
        "fonts": [
            {
                "weight": 100,
                "asset": "fonts/Roboto/Roboto-Thin.ttf"
            },
            {
                "weight": 300,
                "asset": "fonts/Roboto/Roboto-Light.ttf"
            },
            {
                "weight": 400,
                "asset": "fonts/Roboto/Roboto-Regular.ttf"
            },
            {
                "weight": 500,
                "asset": "fonts/Roboto/Roboto-Medium.ttf"
            },
            {
                "weight": 700,
                "asset": "fonts/Roboto/Roboto-Bold.ttf"
            },
            {
                "weight": 900,
                "asset": "fonts/Roboto/Roboto-Black.ttf"
            }
        ]
    },
    {
        "family": "RobotoMono",
        "fonts": [
            {
                "weight": 100,
                "asset": "fonts/Roboto_Mono/RobotoMono-Thin.ttf"
            },
            {
                "weight": 300,
                "asset": "fonts/Roboto_Mono/RobotoMono-Light.ttf"
            },
            {
                "weight": 400,
                "asset": "fonts/Roboto_Mono/RobotoMono-Regular.ttf"
            },
            {
                "weight": 500,
                "asset": "fonts/Roboto_Mono/RobotoMono-Medium.ttf"
            },
            {
                "weight": 700,
                "asset": "fonts/Roboto_Mono/RobotoMono-Bold.ttf"
            }
        ]
    },
    {
        "family": "Octicons",
        "fonts": [
            {
                "asset": "fonts/Octicons.ttf"
            }
        ]
    },
    {
        "family": "Codicon",
        "fonts": [
            {
                "asset": "fonts/codicon.ttf"
            }
        ]
    },
    {
        "family": "packages/codicon/Codicon",
        "fonts": [
            {
                "asset": "packages/codicon/lib/font/codicon.ttf"
            }
        ]
    }
]
```

**Comment by devoncarew** [comment]
It's possible that after editing the pubspec file, we need to re-run pub get in order to re-generate the manifest files?

**Comment by luisgustavoo** [comment]
I'm not sure I understand that question. I deleted the files in the `AppData\Roaming\Pub\Cache ` folder and run the `pub global activate devtools` command again, but the problem persists.

**Comment by kenzieschmoll** [comment]
@luisgustavoo I think Devon was responding to my question
> @devoncarew we are rewriting the pubspec with the changes in #2739 - any idea how we cold still be attempting to load the font from package/codicon/lib?

We could try adding that to the publish script. With the last change, I was unable to verify the fix without a full pub publish, so that may be the case here too. We could either do a +1 dot release or wait to verify with the 2.0.1 release - this isn't causing any real problems with the app itself, so I don't know if it is critical to do an additional release to verify that fix (if we can't verify locally).

**Comment by devoncarew** [comment]
> We could try adding that to the publish script. With the last change, I was unable to verify the fix without a full pub publish, so that may be the case here too. We could either do a +1 dot release or wait to verify with the 2.0.1 release - this isn't causing any real problems with the app itself, so I don't know if it is critical to do an additional release to verify that fix (if we can't verify locally).

If it's not causing an issue then it's probably not worth a publish. If we do do a speculative fix, we could verify it on the next dev release (though, we should be able to verify that the manifest files don't have multiple entries by just inspecting the build/ folder contents).

**Comment by kenzieschmoll** [comment]
Another report of this issue. I'm fairly certain the white page and the failed to load codicon font message are unrelated, though the blank white page on start is still a problem

**Comment by Tilkimen** [comment]
I had the same issue. Solved by changing the 127.0.0.1 part with 'localhost' in the URL. Maybe this might help you to diagnose it. Check here for details: https://github.com/dart-archive/stagehand/issues/620

**Comment by kenzieschmoll** [comment]
Closing as obsolete since this issue is 3+ years old and we have not had any recent reports.
